### PR TITLE
Restrict creating Power VS machine with negative processor and memory

### PR DESCRIPTION
**Pull request by Karthik-K-N**
Related to bug: https://issues.redhat.com/browse/OCPBUGS-1018

### CODE REVIEW

File: pkg/webhooks/machine_webhook.go [hunk 1]
```diff
@@ -1818,8 +1820,9 @@ func validateMachineConfigurations(providerSpec *machinev1.PowerVSMachineProvide
 				errs = append(errs, field.Invalid(parentPath.Child("processor"), processor, fmt.Sprintf("for %s systemtype the maximum supported processor value is %f", providerSpec.SystemType, val.maxProcessor)))
 			}
 		}
-		// validate minimum processor values depending on ProcessorType
-		if providerSpec.ProcessorType == machinev1.PowerVSProcessorTypeDedicated {
+		if processor < 0 {
```
Comment (mkumatag) [inline pkg/webhooks/machine_webhook.go:1823]: This whole block can be written as
```golang
	if processor < 0 {
		errs = append(errs, field.Invalid(parentPath.Child("processor"), processor, "processor value cannot be negative"))
	} else if providerSpec.ProcessorType == machinev1.PowerVSProcessorTypeDedicated && processor < val.minProcessorDedicated {
		warnings = append(warnings, fmt.Sprintf("providerspec.Processor %f is less than the minimum value %f for providerSpec.ProcessorType: %s", processor, val.minProcessorDedicated, providerSpec.ProcessorType))
	} else if processor < val.minProcessorSharedCapped {
		warnings = append(warnings, fmt.Sprintf("providerspec.Processor %f is less than the minimum value %f for providerSpec.ProcessorType: %s", processor, val.minProcessorSharedCapped, providerSpec.ProcessorType))
	}
```

**Review by mkumatag — CHANGES_REQUESTED** [review]
small change, otherwise lgtm

**Review by JoelSpeed — COMMENTED** [review]
Why does being less than the value not catch this? The minimum value should be positive right?

**Comment by Karthik-K-N** [comment]
> Why does being less than the value not catch this? The minimum value should be positive right?

Currently if the user provided value is less than the recommended we just show the warning but not error out the request. So I have made the changes to error out incase the user provided value is negative.

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-operator

Autoscaler test failed, known issue, unrelated to this PR, LGTM

**Pull request merged**